app/services/compute_resource_cache.rb

```diff
@@ -0,0 +1,82 @@
+# This class caches attributes for a compute resource in
+# rails cache to speed up slow or expensive API calls
+class ComputeResourceCache
+  attr_accessor :compute_resource, :cache_duration, :cache_enabled
+  alias_method :cache_enabled?, :cache_enabled
+
+  delegate :logger, :to => ::Rails
+
+  def initialize(compute_resource, opts = {})
+    self.compute_resource = compute_resource
+    self.cache_duration = opts.fetch(:cache_duration, 180.minutes)
```

Comment:
Yes, I do want to add a hourly rake task for this that can be called by cron. I'd prefer to do this in a separate PR, though.